nixos/doc/manual/configuration/fhs.chapter.md

```diff
@@ -0,0 +1,199 @@
+# Binaries from other distributions {#sec-fhs}
+
+Software packaged in Nixpkgs is built to be platform-agnostic in the sense that
+it doesn't expect any dependencies to be implicitly provided by the host
+environment. If a Nix-built program needs to load dynamic shared libraries, the
+concrete revisions of these libraries are always deployed with the program in
+predictable locations. This way, a program built by Nix can be run on any other
+machine with the same architecture, regardless of the environment. For example,
+such a program doesn't need global paths like `/bin` or `/usr/lib` to exist,
+and it always loads the same versions of its dependencies.
+This determinism has known limits, cf. [](#sec-fhs-drivers).
+
+Programs distributed in the binary form outside Nixpkgs will often expect to
+find their dependencies in some of the several common global locations.
+Specifically they would expect to be run in the so-called "merged FHS
+environment" (for [Filesystem Hierarchy
+Standard](https://en.wikipedia.org/wiki/Filesystem_Hierarchy_Standard)). This
+concerns software downloaded using `pip` or `conda install`, toolchains fetched
+by build systems such as `gradle`, shell scripts with hard-coded paths, or
+anything downloaded from the web.
+
+In Nixpkgs the "FHS" hierarchy is splayed, rather than merged: every revision
+of a program is installed into its own prefix. NixOS does not materialize this
+hierarchy into any global paths, meaning that the user wishing to run a program
+expecting the "FHS" environment needs to explicitly point the program at its
+dependencies. NixOS provides several tools that automate or otherwise
+accommodate this task.
+
+## Dynamic linkage issues {#sec-fhs-dynamic-linkage}
+
+Dynamically linked native programs built for "FHS" environments expect to find
+the so-called "dynamic linker" in a fixed location, e.g.
+`/lib64/ld-linux-x86-64.so.2` depending on the platform. They might also expect
```

Comment:
I think it might make sense to describe that nix-built binaries have something like `/nix/store/00000000000000000000000000000000-glibc-2.xx-xx/lib/ld-linux-x86-64.so.2` set as dynamic linker.